quarkus-cli/src/test/java/io/quarkus/ts/quarkus/cli/update/Quarkus213to38CliUpdateIT.java

```diff
@@ -0,0 +1,39 @@
+package io.quarkus.ts.quarkus.cli.update;
+
+import static io.quarkus.test.utils.AwaitilityUtils.untilAsserted;
+
+import java.time.Duration;
+
+import org.apache.http.HttpStatus;
+import org.apache.maven.artifact.versioning.DefaultArtifactVersion;
+import org.junit.jupiter.api.Test;
+
+import io.quarkus.test.bootstrap.QuarkusCliRestService;
+import io.quarkus.test.logging.Log;
+import io.quarkus.test.scenarios.annotations.EnabledOnQuarkusVersion;
+import io.quarkus.test.utils.AwaitilityUtils;
+
+@EnabledOnQuarkusVersion(version = "3.8.*", reason = "This class is testing only updates to 3.8.* versions")
```

Comment:
Does this condition work? As I understand, tests will run only if quarkus version in project is 3.8 stream

Reply:
You understand it right. This test class will run only if testing 3.8 stream. Idea is, that only in this case it has the 3.8 RHBQ repo available.

In all my local trying/testing this condition worked.

Reply:
Well, I missed that. It doesn't really make sense to add to `main` branch test that only run with 3.8 for which we have dedicated branch. We should never run `main` branch with `3.8` so it doesn't makes sense.

Reply:
I didn't realize that. Should I close this PR and merge it only to 3.8? 
I can put into main only the stuff that is not version specific but those are no actual tests.

Reply:
> I didn't realize that. Should I close this PR and merge it only to 3.8?

I think tests specific to 3.8 and lesser as well as code and resources only required by these tests (including test code unless it's planned for reuse) should be only merged to 3.8, yes.

> I can put into main only the stuff that is not version specific but those are no actual tests.

There is 3.9, 3.10, 3.11, 3.12, 3.13 and soon 3.14.  I can see recipes:
- https://github.com/quarkusio/quarkus-updates/blob/main/recipes/src/main/resources/quarkus-updates/core/3.9.alpha1.yaml
- https://github.com/quarkusio/quarkus-updates/blob/main/recipes/src/main/resources/quarkus-updates/core/3.10.alpha1.yaml
- https://github.com/quarkusio/quarkus-updates/blob/main/recipes/src/main/resources/quarkus-updates/core/3.11.yaml
- https://github.com/quarkusio/quarkus-updates/blob/main/recipes/src/main/resources/quarkus-updates/core/3.12.yaml
- https://github.com/quarkusio/quarkus-updates/blob/main/recipes/src/main/resources/quarkus-updates/core/3.13.alpha1.yaml

These migrations will be part of the next LTS, therefore also in RHBQ and customers will use them when migrating between RHBQs. Maybe you could consider testing some of that?

Reply:
> These migrations will be part of the next LTS, therefore also in RHBQ and customers will use them when migrating between RHBQs. Maybe you could consider testing some of that?

 I can do that - is it OK to have this stuff in main?


> I think tests specific to 3.8 and lesser as well as code and resources only required by these tests (including test code unless it's planned for reuse) should be only merged to 3.8, yes.

OK I will rework it into multiple PRs and send 3.8 specific stuff only to 3.8 branch.

Reply:
> I can do that - is it OK to have this stuff in main?

IMO yes.

> OK I will rework it into multiple PRs and send 3.8 specific stuff only to 3.8 branch.

Cool, thanks.

Reply:
In the end, I removed the version specific stuff, it will be another PR into the 3.8 branch. In this PR there is only version-independent stuff. If it is OK I would like to merge this and put tests for 3.8+ updates into separate PR.

Reply:
^^ @michalvavrik

Reply:
ok